examples/multi-agent-debate-system/main.py

```diff
@@ -0,0 +1,299 @@
+        choices=["1", "2", "3", "4"],
+        default="1",
+    )
+    logger.info("Debate topic selected: %s", debate_topic)
+
+    # Selecting the topic and the appropriate Pro or Con agent
+    if debate_topic == "1":
+        student_side = Prompt.ask(
+            "Which side would you like to debate on?\n1. Pro-AI\n2. Con-AI",
+            choices=["1", "2"],
+            default="1",
+        )
+        selected_agent_task = tasks['pro_ai_task'] if student_side == "1" else tasks['con_ai_task']
+        opposing_task = tasks['con_ai_task'] if student_side == "1" else tasks['pro_ai_task']
+    elif debate_topic == "2":
+        student_side = Prompt.ask(
+            "Which side would you like to debate on?\n1. Pro-IP\n2. Con-IP",
+            choices=["1", "2"],
+            default="1",
+        )
+        selected_agent_task = tasks['pro_ip_task'] if student_side == "1" else tasks['con_ip_task']
+        opposing_task = tasks['con_ip_task'] if student_side == "1" else tasks['pro_ip_task']
+    elif debate_topic == "3":
+        student_side = Prompt.ask(
+            "Which side would you like to debate on?\n1. Pro-Bias\n2. Con-Bias",
+            choices=["1", "2"],
+            default="1",
+        )
+        selected_agent_task = tasks['pro_bias_task'] if student_side == "1" else tasks['con_bias_task']
+        opposing_task = tasks['con_bias_task'] if student_side == "1" else tasks['pro_bias_task']
+    elif debate_topic == "4":
+        student_side = Prompt.ask(
+            "Which side would you like to debate on?\n1. Pro-Edu\n2. Con-Edu",
+            choices=["1", "2"],
+            default="1",
+        )
+        selected_agent_task = tasks['pro_edu_task'] if student_side == "1" else tasks['con_edu_task']
+        opposing_task = tasks['con_edu_task'] if student_side == "1" else tasks['pro_edu_task']
+    else:
+        logger.error("Invalid topic selected.")
+        return
+
+    logger.info("Student selected side: %s", "Pro" if student_side == "1" else "Con")
+    student_arguments = []
+    opposing_arguments = []
+    first_turn = True
+    is_pro_turn = True
+    max_turns = 4  # Maximum number of exchanges
+
+    # === Debate Loop ===
+    debate_ended_by_user = False  # Flag to track if the user ended the debate
+
+    for turn in range(max_turns):
+        if llm_delegate:
+            # Autonomous debate mode
+            last_argument = student_arguments[-1] if student_arguments else ""
+            context = f"Opponent's argument: {last_argument}"
+            if is_pro_turn:
+                response = selected_agent_task.run(context)  # Pass context directly
+                if response:
+                    student_arguments.append(response.content)
+                    logger.info("Pro agent's response added.")
+                    print("Pro agent's response added.")
+            else:
+                response = opposing_task.run(context)  # Pass context directly
```

Comment:
Note that each time you use `task.run()` the agent state is reset, so it has no memory of prior arguments (since `Task.__init__()` has `restart=True` by default. In fact you don't even need to use `Task` in this example since you are using a task in single-round mode and there are no tools/function-calls, i.e. you only need the agent's `llm_response` method. So your code will be simpler if you directly use the `agent.llm_response()` method; in this case the agent *does* accumulate conversation each time you call it, unlike using `task.run()`.